### PR TITLE
parser: Added ICF value parsing function.

### PR DESCRIPTION
**Pull request by pepol**
- this function takes value and checks it against all possible types in
   the following order:
   - boolean
   - number
   - quoted string
   - unquoted string
 - note that list values are not yet supported.

Issue: #2 
Depends: #41 
Reviewer: @JakubNvk 
Cc: @milancerny @Anna-Rebeka

### CODE REVIEW

File: parser/icf.py [hunk 1]
```diff
@@ -160,12 +161,34 @@ def _parse_value(value):
...
+    return int(number, 0)
+
+  quoted_match = re.match(RE_QUOTED_STRING, value)
+  if quoted_match is not None:
+    return str(quoted_match.group(1))
+
+  # TODO(pepol): Figure out how to parse lists from unquoted string.
+  unquoted_match = re.match(RE_UNQUOTED_STRING, value)
+  if unquoted_match is not None:
+    # There might be some spaces left over if comment follows so we must take
```
Comment (JakubNvk) [inline parser/icf.py]: I don't think this comment is necessary @pepol

File: parser/icf.py [hunk 2]
```diff
@@ -160,12 +161,34 @@ def _parse_value(value):
...
+    # care to remove them on our own.
```
Comment (JakubNvk) [inline parser/icf.py]: @pepol could you rephrase this? It sounds like we'd have to manually delete the spaces.

Comment (pepol) [inline parser/icf.py]: Done.

**Comment by JakubNvk** [comment]
:+1: @pepol please squash so I can merge :)
